beacon/beacon_network.go

```diff
@@ -353,3 +359,68 @@ func (bn *Network) stateSummariesValidation(f ForkedHistoricalSummariesWithProof
 	gIndex := 59
 	return merkle.VerifyMerkleBranch(summariesRoot, proof.Proof[:], 5, uint64(gIndex), latestFinalizedRoot)
 }
+
+func (bn *Network) GetHeadHash() (*gcommon.Hash, error) {
```

Comment:
I think we could not make `Network` struct has a `bn` dependency, and create a interface about `getHead` or `Validate` which used different provider such as beacon, external or others.